[anlamm] iP

anlamm opened the pull request.

anlamm committed:
Level 0. Rename, Greet, Exit

Rename to Anke, added greet and exit

anlamm committed:
Stop tracing Duke

anlamm committed:
Level 1. Echo

anlamm committed:
Level 2. Add, List

anlamm committed:
Level 3. Mark as Done

added Task class to store the name and status of each task

anlamm committed:
Level 4. ToDos, Events, Deadlines

Add support for tracking three types of tasks:

ToDos: tasks without any date/time attached to it e.g., visit new theme park
Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
Events: tasks that start at a specific date/time and ends at a specific date/time
e.g., (a) team project meeting 2/10/2019 2-4pm (b) orientation week 4/10/2019 to 11/10/2019

jemehgoh commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,66 @@
+            } else if (line.length() > 6 && line.substring(0, 6).equals("event ")) {
+                int fromIndex = line.indexOf("/from");
+                int toIndex = line.indexOf("/to");
+                task = new Event(line.substring(6, fromIndex - 1), line.substring(fromIndex + 6, toIndex - 1), line.substring(toIndex + 4));
```
Could wrap this line and put a line break after the second call to line.substring(), as per the coding standard?

jemehgoh commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,66 @@
+                        continue;
+                    }
+                }
+                catch (NumberFormatException e) {
```
Could you change the formatting of this statement, such that the closing brace and the catch() condition are on the same line?

jemehgoh commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,66 @@
+                System.out.println("");
+                continue;
+            } else if (line.length() > 5 && line.substring(0, 5).equals("mark ")) {
+                int index;
```
Could you declare the "index" variable within the try block, in the same line as the initialisation (line 25)?

jemehgoh commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,66 @@
+                }
+                catch (NumberFormatException e) {
+                }
+            } else if (line.length() > 5 && line.substring(0, 5).equals("todo ")) {
+                task = new Todo(line.substring(5));
+            } else if (line.length() > 9 && line.substring(0, 9).equals("deadline ")) {
+                int byIndex = line.indexOf("/by");
+                task = new Deadline(line.substring(9, byIndex - 1), line.substring(byIndex + 4));
+            } else if (line.length() > 6 && line.substring(0, 6).equals("event ")) {
+                int fromIndex = line.indexOf("/from");
+                int toIndex = line.indexOf("/to");
```
I like your indentation and your casing for the fromIndex and toIndex variables

TrungBui32 reviewed (commented):
Overall, good code!

jemehgoh reviewed (commented):
All in all, most of the code follows the coding standards, with only a few violations.

anlamm committed:
A-CodeQuality, A-TextUiTesting

anlamm committed:
handling exceptions

anlamm committed:
handle exceptions for mark and unmark

anlamm committed:
Merge branch 'branch-Level-5'

anlamm committed:
level-6 delete
A-Collections

Add support for deleting tasks from the list with the use of Collections class

anlamm committed:
Level 7. Save

anlamm committed:
Merge branch 'branch-Level-6'

anlamm committed:
Merge branch 'branch-Level-7'

resolved merge conflict

anlamm committed:
fixed loading bugs

tzerbin commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,316 @@
+    static Scanner in = new Scanner(System.in);
+
+    public static void printWelcome() {
+        System.out.println("Hello! I'm Anke.");
```
Perhaps refrain from using magic strings and numbers in the entire codebase!

tzerbin commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,316 @@
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Anke {
```
Perhaps wrap classes in packages for the entire codebase?

tzerbin replied:
This is a very long piece of code. We will be looking at class design some time down the line which may mitigate this situation.

tzerbin commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,316 @@
+import java.io.IOException;
+
+public class Anke {
+    static int count = 0;
```
Could you be more descriptive with the variable here?

tzerbin commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,316 @@
+        printBye();
+    }
+
+    private static void load(ArrayList<Task> tasks) {
```
Could you be more descriptive with the method name?

tzerbin commented on src/main/java/Anke.java:
```diff
@@ -0,0 +1,316 @@
+        System.out.println("Now you have " + count + " tasks in the list.\n");
+    }
+
+    private static int getIndex(ArrayList<Task> tasks, String line, int beginIndex, String func) throws NumberFormatException, IndexOutOfRangeException, TaskSameStateException {
```
The method signature is long, perhaps use line breaks for this?

tzerbin reviewed (commented):
Good attempt in following guidelines!

However, almost all your code are within one file, which you should break up for better readability. We will be going through class designs really soon.

Also, there are a number of critical issues such as not wrapping classes in packages, descriptive variable names and usage of magic strings/numbers.

anlamm committed:
MoreOOP

Make the code more OOP

anlamm committed:
Level 9. Find

Give users a way to find a task by searching for a keyword.

anlamm committed:
Merge pull request #1 from anlamm/branch-Level-9

Level 9. Find

anlamm committed:
A-JavaDoc

 Add JavaDoc comments

anlamm committed:
Merge pull request #2 from anlamm/A-JavaDoc

A-JavaDoc

anlamm committed:
improve comments

anlamm committed:
Debug and modified readme file

anlamm committed:
update readme
